src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Android.cs

```diff
@@ -0,0 +1,762 @@
+		{
+			int indexSize   = dataOffset - indexOffset;
+			int entryCount  = indexSize / Marshal.SizeOf (typeof (AndroidTzDataEntry));
+			int entrySize   = Marshal.SizeOf (typeof (AndroidTzDataEntry));
+
+			byteOffsets   = new int [entryCount];
+			ids           = new string [entryCount];
+			lengths       = new int [entryCount];
+
+			for (int i = 0; i < entryCount; ++i) {
+				var entry = ReadAt<AndroidTzDataEntry>(indexOffset + (entrySize*i), buffer);
+				var p     = (sbyte*) entry.id;
+
+				byteOffsets [i]   = NetworkToHostOrder (entry.byteOffset) + dataOffset;
+				ids [i]           = new string (p, 0, GetStringLength (p, 40), Encoding.ASCII);
+				lengths [i]       = NetworkToHostOrder (entry.length);
+
+				if (lengths [i] < Marshal.SizeOf (typeof (AndroidTzDataHeader)))
+					throw new InvalidOperationException ("Length in index file < sizeof(tzhead)");
+			}
+		}
+
+		static unsafe int GetStringLength (sbyte* s, int maxLength)
+		{
+			int len;
+			for (len = 0; len < maxLength; len++, s++) {
+				if (*s == 0)
+					break;
+			}
+			return len;
+		}
+
+		unsafe void ReadZoneTab (int zoneTabOffset, int zoneTabSize)
+		{
+			byte[] zoneTab = new byte [zoneTabSize];
+
+			data.Position = zoneTabOffset;
+
+			int r;
+			if ((r = data.Read (zoneTab, 0, zoneTab.Length)) < zoneTab.Length)
+				throw new InvalidOperationException (
+						string.Format ("Error reading zonetab: read {0} bytes, expected {1}", r, zoneTabSize));
+
+			this.zoneTab = Encoding.ASCII.GetString (zoneTab, 0, zoneTab.Length);
+		}
+
+		public IEnumerable<string> GetAvailableIds ()
+		{
+			return ids;
+		}
+
+		public byte[] GetTimeZoneData (string id)
+		{
+			int i = Array.BinarySearch (ids, id, StringComparer.Ordinal);
+			if (i < 0)
+				return null;
+
+			int offset = byteOffsets [i];
+			int length = lengths [i];
+			var buffer = new byte [length];
+
+			lock (data) {
+				data.Position = offset;
+				int r;
+				if ((r = data.Read (buffer, 0, buffer.Length)) < buffer.Length)
+					throw new InvalidOperationException (
+							string.Format ("Unable to fully read from file '{0}' at offset {1} length {2}; read {3} bytes expected {4}.",
+								tzdataPath, offset, length, r, buffer.Length));
+			}
+
+			TimeZoneInfo.DumpTimeZoneDataToFile (id, buffer);
+			return buffer;
+		}
+	}
+
+	partial class TimeZoneInfo {
+
+		static TimeZoneInfo CreateLocal ()
+		{
+			return AndroidTimeZones.Local;
+		}
+
+		static TimeZoneInfo FindSystemTimeZoneByIdCore (string id)
+		{
+			var timeZoneInfo = AndroidTimeZones.GetTimeZone (id, id);
+			if (timeZoneInfo == null)
+				throw new TimeZoneNotFoundException ();
+			return timeZoneInfo;
+		}
+
+		static void GetSystemTimeZonesCore (List<TimeZoneInfo> systemTimeZones)
+		{
+			foreach (string id in AndroidTimeZones.GetAvailableIds ()) {
+				var tz = AndroidTimeZones.GetTimeZone (id, id);
+				if (tz != null)
+					systemTimeZones.Add (tz);
+			}
+		}
+
+		/*
+		 * Android < v4.3 Timezone support infrastructure.
```

Comment:
we don't support Android < v4.3 in dotnet/runtime anymore so all of this code can probably go away